samples/sid_end_device/src/sidewalk.c

```diff
@@ -29,16 +32,32 @@ static void sid_thread_entry(void *context, void *unused, void *unused2)
 
 	while (1) {
 		int err = k_msgq_get(&sidewalk_thread_msgq, &event, K_FOREVER);
-		if (!err) {
+		switch (err) {
+		case 0: {
+			LOG_DBG("event received %p (%s) sidewalk workq usage (%d/%d) ( after get )",
+				(void *)(event.handler), EVENT_TO_NAME(event.handler),
+				k_msgq_num_used_get(&sidewalk_thread_msgq),
+				CONFIG_SIDEWALK_THREAD_QUEUE_SIZE);
 			if (event.handler) {
 				event.handler(sid, event.ctx);
 			}
 			if (event.ctx_free) {
 				event.ctx_free(event.ctx);
 			}
-		} else {
+			if (event.handler != sidewalk_event_process) {
+				sidewalk_event_process(sid, NULL);
```

Comment:
I don't understand this. Most events triggers sid process, and it will be added to msgq anyway. Moreover it adds extra load to queue.